packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -164,43 +164,14 @@ def __new__(mcs, name: str, bases: Tuple, namespace: Dict, **kwargs: Any) -> Typ
         if ABC in bases:
             # abstract class, return
             return new_cls
-        if issubclass(new_cls, BaseTxPayload):
-            mcs.transaction_type_to_payload_cls[new_cls.__name__] = new_cls  # type: ignore
-            return new_cls
         if not issubclass(new_cls, BaseTxPayload):
             raise ValueError(  # pragma: no cover
                 f"class {name} must inherit from {BaseTxPayload.__name__}"
             )
-        new_cls = cast(Type[BaseTxPayload], new_cls)
-
-        transaction_type = str(mcs._get_field(new_cls, "transaction_type"))
-        mcs._validate_transaction_type(transaction_type, new_cls)
-        # remember association from transaction type to payload class
-        mcs.transaction_type_to_payload_cls[transaction_type] = new_cls
 
+        mcs.transaction_type_to_payload_cls[new_cls.__name__] = new_cls
         return new_cls
 
-    @classmethod
-    def _validate_transaction_type(
-        mcs, transaction_type: str, new_payload_cls: Type["BaseTxPayload"]
-    ) -> None:
-        """Check that a transaction type is not already associated to a concrete payload class."""
-        if transaction_type in mcs.transaction_type_to_payload_cls:
-            previous_payload_cls = mcs.transaction_type_to_payload_cls[transaction_type]
-            if new_payload_cls.__name__ != previous_payload_cls.__name__:
-                raise ValueError(
-                    f"transaction type with name {transaction_type} already "
-                    f"used by class {previous_payload_cls}, and cannot be "
-                    f"used by class {new_payload_cls} "
-                )
```

Comment:
So there either is an issue in the current design or I misunderstand it. Now `transaction_type` and `payload.__name__` need to match. However, that allows for an overwrite which I believe should not happen. To illustrate, the following is possible:

```
class Payload(BaseTxPayload):
    transaction_type = "tx_type"

class Payload(BaseTxPayload):
    transaction_type = "tx_type"
    other_field = "other"
```